src/Halogen/Canvas/Animate.purs

```diff
@@ -0,0 +1,89 @@
+module Halogen.Canvas.Animate
+  ( component
+  ) where
+
+import Prelude
+
+import Control.Monad.Reader (runReaderT)
+import Control.Monad.Rec.Class (class MonadRec)
+import Data.Maybe (Maybe(..))
+import Data.Traversable (traverse_)
+import Effect.Aff.Class (class MonadAff)
+import Graphics.Canvas (getContext2D)
+import Graphics.Canvas.Free (CanvasContext, CanvasT, clearRect, getHeight, getWidth, runCanvasT, withContext)
+import Halogen as H
+import Halogen.HTML as HH
+import Halogen.HTML.Properties as HP
+import Halogen.Subscription as HS
+import Unsafe.Coerce (unsafeCoerce)
+import Web.HTML (window)
+import Web.HTML.Window.AnimationFrame (DOMHighResTimestamp, requestAnimationFrame)
+
+type Dimensions =
+  { width :: Int
+  , height :: Int
+  }
+
+
+type Animation m =
+  { dimensions :: Dimensions
+  , animation :: DOMHighResTimestamp -> CanvasT m Unit
+  }
```

Comment:
The notion is that this is [similar to a Gloss event loop (specifically "animate") except the picture is drawn in an imperative style in `CanvasT`](https://hackage.haskell.org/package/gloss-1.13.2.2/docs/Graphics-Gloss.html).